### PR TITLE
backrest: init at 1.5.0

### LINKED ISSUE #312298: Package request: backrest
**Project description**

Backrest is a web-accessible backup solution built on top of [restic](https://restic.net/). Backrest provides a WebUI which wraps the restic CLI and makes it easy to create repos, browse snapshots, and restore files. Additionally, Backrest can run in the background and take an opinionated approach to scheduling snapshots and orchestrating repo health operations.
**Metadata**

* homepage URL:
* source URL: https://github.com/garethgeorge/backrest
* license: gpl3
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by interdependence**
## Description of changes

Add [Backrest](https://github.com/garethgeorge/backrest): A web-accessible backup solution built on top of restic.

Closes #312298 @stv0g

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ba/backrest/package.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  # Fails with handler returned wrong content encoding
+  checkFlags = [ "-skip=TestServeIndex" ];
+
+  preCheck = ''
+    # Use restic from nixpkgs, otherwise download fails in sandbox
+    export BACKREST_RESTIC_COMMAND="${restic}/bin/restic"
+    export HOME=$(pwd)
+  '';
+
+  meta = with lib; {
```
Comment (stv0g) [inline pkgs/by-name/ba/backrest/package.nix:56]: Please try to avoid using `with lib` here as it pulls in a large scope.

**Review by stv0g — CHANGES_REQUESTED** [review]
LGTM with exception of a smal nit

File: pkgs/by-name/ba/backrest/package.nix (hunk 1 above)
Comment (interdependence) [reply pkgs/by-name/ba/backrest/package.nix]: Okay I fixed that.

**Review by stv0g — APPROVED** [review]

File: pkgs/by-name/ba/backrest/package.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  util-linux,
+}:
+let
+  pname = "backrest";
+  version = "1.5.0";
+
+  src = fetchFromGitHub {
+    owner = "garethgeorge";
+    repo = "backrest";
+    rev = "v${version}";
```
[the remark this answers is not in the document]
Comment (interdependence) [reply pkgs/by-name/ba/backrest/package.nix]: Fixed :+1:

File: pkgs/by-name/ba/backrest/package.nix [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    homepage = "https://github.com/garethgeorge/backrest";
+    changelog = "https://github.com/garethgeorge/backrest/releases/tag/v${version}";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ interdependence ];
+    mainProgram = "backrest";
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
```
[the remark this answers is not in the document]
Comment (interdependence) [reply pkgs/by-name/ba/backrest/package.nix]: Fixed

**Review by stv0g — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**